File: src/backend/DonkeyWork.Chat.Api/Models/ApiKey/GetApiKeysModel.cs
// ------------------------------------------------------
// <copyright file="GetApiKeysModel.cs" company="DonkeyWork.Dev">
// Provided as is by (c) DonkeyWork.Dev. No warranty or support is given.
// </copyright>
// ------------------------------------------------------

namespace DonkeyWork.Chat.Api.Models.ApiKey;

/// <summary>
/// A model for api keys.
/// </summary>
public record GetApiKeysModel
{
    /// <summary>
    /// Gets the count of api keys.
    /// </summary>
    public int Count { get; init; }

    /// <summary>
    /// Gets the api keys.
    /// </summary>
    public IEnumerable<ApiKeySummaryModel> ApiKeys { get; init; } = [];
}

File: src/ui/src/models/api/ApiKey/GetApiKeysModel.ts
import { ApiKeySummaryModel } from './ApiKeySummaryModel';

/**
 * Model for retrieving API keys
 */
export interface GetApiKeysModel {
  /**
   * Total count of API keys
   */
  count: number;

  /**
   * Collection of API key summaries
   */
  apiKeys: ApiKeySummaryModel[];
}